File: coordinatorlayout/src/main/java/com/demo/coordinatorlayout/MainActivity.java
package com.demo.coordinatorlayout;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    private RecyclerView recyclerview;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        recyclerview = (RecyclerView) findViewById(R.id.rcl_view_goods);

        List<String> mList=new ArrayList<>();
        for(int i=0;i<30;i++){
            mList.add("item"+i);
        }
        DataAdapter dataAdapter=new DataAdapter(this,mList);
        LinearLayoutManager linearLayoutManager= new LinearLayoutManager(this);
        linearLayoutManager.setOrientation(RecyclerView.VERTICAL);
        recyclerview.setLayoutManager(linearLayoutManager);
        recyclerview.setAdapter(dataAdapter);
    }
}

File: app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import java.util.regex.Matcher
import java.util.regex.Pattern

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val str = "19525558642"
        val pattern = "^1(34[0-8]|3[5-9]\\\\d|4[7-8]\\\\d|5[0-27-9]\\\\d|7[28]\\\\d|8[2-4,7-8]\\\\d|98\\\\d)\\\\d{7}\$";

        val r: Pattern = Pattern.compile(pattern)
        val m: Matcher = r.matcher(str)
        System.out.println("是否移动："+m.matches())
    }
}